### PR TITLE
Use ListingTable for deployments overview

### PR DESCRIPTION
**Pull request by marusak**

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -369,52 +288,176 @@ const DeploymentVersion = ({ info, packages }) => {
...
+        _setError(cpy);
+    };
+
+    const doRollback = (key, osname) => {
+        const args = {
+            reboot: cockpit.variant("b", true)
+        };
+        setInProgress(key, true);
+        return client.run_transaction("Rollback", [args], osname)
+                .catch(ex => setError(key, { title: _("Failed to rollback daployment"), details: ex }))
```
Comment (KKoukiou) [inline src/app.jsx:319]: s/daployment/deployment

**Review by KKoukiou — COMMENTED** [review]
Mobile is not perfect. Tried the pixel5, iphone xr and 12 from the options in the browser.
127 0 0 2_9091_updates(Pixel 5)

**Comment by marusak** [comment]
@KKoukiou Updated your comments
Looks like this on wider screen:
Screenshot from 2023-02-10 15-56-20

And then on mobile:
Screenshot from 2023-02-10 15-57-32

File: src/app.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (marusak) [reply src/app.jsx]: > Roll back is the verb form. Rollback is a noun. Incidentally, the same holds true for log in and login. Also work out and workout.

So yes, it should be `Failed to roll back`

[the remark this answers is not in the document]
Comment (garrett) [reply src/app.jsx]: It's "roll back" when used like this.

**Comment by garrett** [comment]
1. It's supposed to automatically check for updates, right? If so, then checking for updates is redundant and only a workaround, so it should be a refresh icon (but accessible with some text explaining it that should also show up as a tooltip). This is also what we do on the normal updates page.
2. The table is missing either a header or a top line. Entry rows should not lack a top line, one way or another.
3. There isn't enough padding around the button in the row. It's missing a few vertical pixels.
4. There isn't enough information in rows in the table to make a decision. Which one is newer? Which branch is it? etc. Why is "available" below "running", if it's newer? (Is "available" even newer?) Should something even be called "available" if that's the default state?
5. Why are these "deployments"? Don't we also list other available builds? (Mockups have "Deployments & updates" for example.)
 
Also: Is the warning actually a breaking error? Or is it a warning where you can proceed? It sounds like it's a dealbraker, but doesn't look like it. Is it actionable? What can you do to resolve it?

**Comment by marusak** [comment]
> It's supposed to automatically check for updates, right? If so, then checking for updates is redundant and only a workaround, so it should be a refresh icon (but accessible with some text explaining it that should also show up as a tooltip). This is also what we do on the normal updates page.

How do you mean? Like periodically checking for them while page open? Or check once when the page is open? I don't think we do either of these things. Also I just moved the button from one panel to another without any functional changes.

> The table is missing either a header or a top line. Entry rows should not lack a top line, one way or another.

I was looking around how it looks like on other places. For example selinux has top line in one but not in another card:
Screenshot from 2023-02-13 11-23-41

I think I will add header (as we want to show more fields anyway). But I was checking machines and we don't give any name to the last column. Should I show there `Actions` or don't give it any name?

> There isn't enough padding around the button in the row. It's missing a few vertical pixels.

Yeah, I think so as well, just have no idea how to convince it without any custom css. It seems to work fine in machines. Maybe @KKoukiou knows?

> There isn't enough information in rows in the table to make a decision. Which one is newer? Which branch is it? etc. Why is "available" below "running", if it's newer? (Is "available" even newer?) Should something even be called "available" if that's the default state?

This is the current state just using different components. I am not adding any new functionality in this PR and plan that as follow ups.

> Why are these "deployments"? Don't we also list other available builds? (Mockups have "Deployments & updates" for example.)

Will rename to "Deployments and updates"

**Comment by garrett** [comment]
> How do you mean? Like periodically checking for them while page open? Or check once when the page is open?

Fedora IoT, Fedora CoreOS, RHEL Automotive, etc. are all supposed to automatically fetch available changes and even sometimes update (and reboot, etc.).

Silverblue does not, but I'm not suggesting getting rid of refresh, but making it a visible icon only, just like with the software updates page.

(Silverblue is not a server system, but people do use Cocktpit and Silverblue on Silverblue-based systems for development (including us), and I've seen some people (not us) using Cockpit-OStree as a UI for their desktop systems, as it's still one of the few UIs for it.)

> I was looking around how it looks like on other places. For example selinux has top line in one but not in another card

Then that's a bug too.

It should either have a separator line or a separator line plus headings above it. It does not have a top line between the rest of the card and the table, as we do not want a separator line above the headings.

> But I was checking machines and we don't give any name to the last column. Should I show there Actions or don't give it any name?

We usually don't give it a visible name of actions, but it should still have "Actions" as a `aria-label` in the `th` for screen readers, along with the usual scope="col" in the headings (and scope="row" for the named element per row, as shown) @ https://www.w3.org/WAI/tutorials/tables/two-headers/ (Yes, I'm talking about how the HTML should end up being. Adapt it to JSX, of course.)

> This is the current state just using different components. I am not adding any new functionality in this PR and plan that as follow ups.

OK!

> Will rename to "Deployments and updates"

Thanks!

**Comment by marusak** [comment]
Looks like this now:

Screenshot from 2023-02-13 16-29-13
Screenshot from 2023-02-13 16-28-44

**Comment by marusak** [comment]
> I'm not suggesting getting rid of refresh, but making it a visible icon only, just like with the software updates page.

Ok, let's do that in a follow up. There will be many more follow ups :) I have the designs open and go piece by piece through it

**Comment by garrett** [comment]
New mockup, some of which applies here:

OStree page(3)


Also posted on the redesign issue @ 
- https://github.com/cockpit-project/cockpit-ostree/issues/176#issuecomment-1428184241

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**